### PR TITLE
Update runtime v1.2.8

### PR DESCRIPTION
**Pull request by raynaudoe**
- Update relay chain to v1.2.8
- Update Asset Hub to v1.2.8
- Update Bridge Hub to v1.2.8

### CODE REVIEW

**Comment by hbulgarini** [comment]
```
➜  runtimes git:(release/1.2.8-runtime) try-runtime \
--runtime target/release/wbuild/paseo-runtime/paseo_runtime.compact.compressed.wasm  \
    on-runtime-upgrade \
    live --uri wss://paseo.rpc.amforc.com:443
[2024-09-08T20:35:26Z INFO  remote-ext] replacing wss:// in uri with https://: "https://paseo.rpc.amforc.com:443" (ws is currently unstable for fetching remote storage, for more see https://github.com/paritytech/jsonrpsee/issues/1086)
[2024-09-08T20:35:27Z INFO  remote-ext] since no at is provided, setting it to latest finalized head, 0x657a200f1f10ddd57d46414f58f8c770fcf8460afa87835c81f17f77534aaf0c
[2024-09-08T20:35:27Z INFO  remote-ext] since no prefix is filtered, the data for all pallets will be downloaded
[2024-09-08T20:35:27Z INFO  remote-ext] scraping key-pairs from remote at block height 0x657a200f1f10ddd57d46414f58f8c770fcf8460afa87835c81f17f77534aaf0c
✅ Found 94643 keys (3.98s)
[00:00:09] Downloading key values 7,946.4238/s [==========================================================================================================================>-------------------------] 78308/94643 (2s)[2024-09-08T20:35:41Z WARN  frame_remote_externalities] Batch request failed (2/12 retries). Error: Parse error: invalid type: map, expected a sequence at line 1 column 0
[00:00:17] ✅ Downloaded key values 5,511.7978/s [==================================================================================================================================================] 94643/94643 (0s)
✅ Inserted keys into DB (0.33s)
[2024-09-08T20:35:48Z INFO  remote-ext] adding data for hashed prefix: , took 21.63s
[2024-09-08T20:35:48Z INFO  remote-ext] adding data for hashed key: 3a636f6465
[2024-09-08T20:35:49Z INFO  remote-ext] adding data for hashed key: 26aa394eea5630e07c48ae0c9558cef7f9cce9c888469bb1a0dceaa129672ef8
[2024-09-08T20:35:49Z INFO  remote-ext] adding data for hashed key: 26aa394eea5630e07c48ae0c9558cef702a5c1b19ab7a04f536c519aca4983ac
[2024-09-08T20:35:49Z INFO  remote-ext] 👩‍👦 no child roots found to scrape
[2024-09-08T20:35:49Z INFO  remote-ext] initialized state externalities with storage root 0x81cf5306f82db0f8f9536852f448d933d1ee72428ccec57d4bdc1b3cd29950ca and state_version V1
[2024-09-08T20:35:49Z INFO  try-runtime::cli] Original runtime [Name: RuntimeString::Owned("paseo")] [Version: 1002007] [Code hash: 0xa42b...d756]
[2024-09-08T20:35:49Z INFO  try-runtime::cli] New runtime      [Name: RuntimeString::Owned("paseo")] [Version: 1002008] [Code hash: 0xf57c...a480]
[2024-09-08T20:35:49Z INFO  try-runtime::cli] 🚀 Speed up your workflow by using snapshots instead of live state. See `try-runtime create-snapshot --help`.
[2024-09-08T20:35:49Z INFO  try_runtime_core::misc] ------------------------------------------------------------------


[2024-09-08T20:35:49Z INFO  try_runtime_core::misc] 🔬 Running TryRuntime_on_runtime_upgrade with checks: PreAndPost


[2024-09-08T20:35:49Z INFO  try_runtime_core::misc] ------------------------------------------------------------------


[2024-09-08T20:35:50Z INFO  runtime::paseo] try-runtime::on_runtime_upgrade polkadot.
[2024-09-08T20:35:50Z INFO  runtime::frame-support] 🐥 New pallet "ImOnline" detected in the runtime. The pallet has no defined storage version, so the on-chain version is being initialized to StorageVersion(0).
[2024-09-08T20:35:50Z WARN  paseo_runtime::init_state_migration] MigrationProcess is not Default. Not proceeding.
[2024-09-08T20:35:50Z WARN  runtime::session_keys] Skipping session keys migration pre-upgrade check due to spec version (already applied?)
[2024-09-08T20:35:50Z INFO  paseo_runtime::migrations] Skipping session keys upgrade: already applied
[2024-09-08T20:35:50Z WARN  runtime::session_keys] Skipping session keys migration post-upgrade check due to spec version (already applied?)
[2024-09-08T20:35:50Z WARN  frame_support::migrations] 🚚 Pallet "NominationPools" VersionedMigration migration 7->8 can be removed; on-chain is already at StorageVersion(8).
[2024-09-08T20:35:50Z WARN  runtime::staking] [2868541] 💸 v14 not applied.
[2024-09-08T20:35:50Z WARN  frame_support::migrations] 🚚 Pallet "Configuration" VersionedMigration migration 9->10 can be removed; on-chain is already at StorageVersion(11).
[2024-09-08T20:35:50Z WARN  frame_support::migrations] 🚚 Pallet "Configuration" VersionedMigration migration 10->11 can be removed; on-chain is already at StorageVersion(11).
[2024-09-08T20:35:50Z WARN  frame_support::migrations] 🚚 Pallet "Grandpa" VersionedMigration migration 4->5 can be removed; on-chain is already at StorageVersion(5).
[2024-09-08T20:35:50Z WARN  frame_support::migrations] 🚚 Pallet "Identity" VersionedMigration migration 0->1 can be removed; on-chain is already at StorageVersion(1).
[2024-09-08T20:35:50Z WARN  frame_support::migrations] 🚚 Pallet "ParaScheduler" VersionedMigration migration 1->2 can be removed; on-chain is already at StorageVersion(2).
[2024-09-08T20:35:50Z INFO  frame_support::migrations] Found ImOnline keys pre-removal 👀
[2024-09-08T20:35:50Z INFO  frame_support::migrations] Removed 0 ImOnline keys 🧹
[2024-09-08T20:35:50Z INFO  frame_support::migrations] No ImOnline keys found post-removal 🎉
[2024-09-08T20:35:50Z INFO  runtime::executive] ✅ Entire runtime state decodes without error. 44280178 bytes total.
[2024-09-08T20:35:51Z INFO  try_runtime_core::misc] ------------------------------------------------------------------------------------------------------


[2024-09-08T20:35:51Z INFO  try_runtime_core::misc] 🔬 TryRuntime_on_runtime_upgrade succeeded! Running it again without checks for weight measurements.


[2024-09-08T20:35:51Z INFO  try_runtime_core::misc] ------------------------------------------------------------------------------------------------------


[2024-09-08T20:35:51Z INFO  runtime::paseo] try-runtime::on_runtime_upgrade polkadot.
[2024-09-08T20:35:51Z INFO  runtime::frame-support] 🐥 New pallet "ImOnline" detected in the runtime. The pallet has no defined storage version, so the on-chain version is being initialized to StorageVersion(0).
[2024-09-08T20:35:51Z WARN  paseo_runtime::init_state_migration] MigrationProcess is not Default. Not proceeding.
[2024-09-08T20:35:51Z INFO  paseo_runtime::migrations] Skipping session keys upgrade: already applied
[2024-09-08T20:35:51Z WARN  frame_support::migrations] 🚚 Pallet "NominationPools" VersionedMigration migration 7->8 can be removed; on-chain is already at StorageVersion(8).
[2024-09-08T20:35:51Z WARN  runtime::staking] [2868541] 💸 v14 not applied.
[2024-09-08T20:35:51Z WARN  frame_support::migrations] 🚚 Pallet "Configuration" VersionedMigration migration 9->10 can be removed; on-chain is already at StorageVersion(11).
[2024-09-08T20:35:51Z WARN  frame_support::migrations] 🚚 Pallet "Configuration" VersionedMigration migration 10->11 can be removed; on-chain is already at StorageVersion(11).
[2024-09-08T20:35:51Z WARN  frame_support::migrations] 🚚 Pallet "Grandpa" VersionedMigration migration 4->5 can be removed; on-chain is already at StorageVersion(5).
[2024-09-08T20:35:51Z WARN  frame_support::migrations] 🚚 Pallet "Identity" VersionedMigration migration 0->1 can be removed; on-chain is already at StorageVersion(1).
[2024-09-08T20:35:51Z WARN  frame_support::migrations] 🚚 Pallet "ParaScheduler" VersionedMigration migration 1->2 can be removed; on-chain is already at StorageVersion(2).
[2024-09-08T20:35:51Z INFO  frame_support::migrations] Removed 0 ImOnline keys 🧹
[2024-09-08T20:35:51Z INFO  try_runtime_core::misc] ---------------------------------------------------------------------------------


[2024-09-08T20:35:51Z INFO  try_runtime_core::misc] 🔬 Running TryRuntime_on_runtime_upgrade again to check idempotency: PreAndPost


[2024-09-08T20:35:51Z INFO  try_runtime_core::misc] ---------------------------------------------------------------------------------


[2024-09-08T20:35:51Z INFO  runtime::paseo] try-runtime::on_runtime_upgrade polkadot.
[2024-09-08T20:35:51Z INFO  runtime::frame-support] 🐥 New pallet "ImOnline" detected in the runtime. The pallet has no defined storage version, so the on-chain version is being initialized to StorageVersion(0).
[2024-09-08T20:35:51Z WARN  paseo_runtime::init_state_migration] MigrationProcess is not Default. Not proceeding.
[2024-09-08T20:35:51Z WARN  runtime::session_keys] Skipping session keys migration pre-upgrade check due to spec version (already applied?)
[2024-09-08T20:35:51Z INFO  paseo_runtime::migrations] Skipping session keys upgrade: already applied
[2024-09-08T20:35:51Z WARN  runtime::session_keys] Skipping session keys migration post-upgrade check due to spec version (already applied?)
[2024-09-08T20:35:51Z WARN  frame_support::migrations] 🚚 Pallet "NominationPools" VersionedMigration migration 7->8 can be removed; on-chain is already at StorageVersion(8).
[2024-09-08T20:35:51Z WARN  runtime::staking] [2868541] 💸 v14 not applied.
[2024-09-08T20:35:51Z WARN  frame_support::migrations] 🚚 Pallet "Configuration" VersionedMigration migration 9->10 can be removed; on-chain is already at StorageVersion(11).
[2024-09-08T20:35:51Z WARN  frame_support::migrations] 🚚 Pallet "Configuration" VersionedMigration migration 10->11 can be removed; on-chain is already at StorageVersion(11).
[2024-09-08T20:35:51Z WARN  frame_support::migrations] 🚚 Pallet "Grandpa" VersionedMigration migration 4->5 can be removed; on-chain is already at StorageVersion(5).
[2024-09-08T20:35:51Z WARN  frame_support::migrations] 🚚 Pallet "Identity" VersionedMigration migration 0->1 can be removed; on-chain is already at StorageVersion(1).
[2024-09-08T20:35:51Z WARN  frame_support::migrations] 🚚 Pallet "ParaScheduler" VersionedMigration migration 1->2 can be removed; on-chain is already at StorageVersion(2).
[2024-09-08T20:35:51Z INFO  frame_support::migrations] Found ImOnline keys pre-removal 👀
[2024-09-08T20:35:51Z INFO  frame_support::migrations] Removed 0 ImOnline keys 🧹
[2024-09-08T20:35:51Z INFO  frame_support::migrations] No ImOnline keys found post-removal 🎉
[2024-09-08T20:35:51Z INFO  runtime::executive] ✅ Entire runtime state decodes without error. 44280178 bytes total.
[2024-09-08T20:35:51Z INFO  try-runtime::cli] PoV size (zstd-compressed compact proof): 9.5 KB. For parachains, it's your responsibility to verify that a PoV of this size fits within any relaychain constraints.
[2024-09-08T20:35:51Z INFO  try-runtime::cli] Consumed ref_time: 0.00164287s (0.08% of max 2s)
[2024-09-08T20:35:51Z INFO  try-runtime::cli] ✅ No weight safety issues detected. Please note this does not guarantee a successful runtime upgrade. Always test your runtime upgrade with recent state, and ensure that the weight usage of your migrations will not drastically differ between testing and actual on-chain execution.
```

**Review by hbulgarini — APPROVED** [review]

**Review by al3mart — COMMENTED** [review]
It seems that we still have a couple of references to Kusama here and there.
I think we would want to remove them or substitute them for a Westend config instead.

File: system-parachains/asset-hub-paseo/src/lib.rs [hunk 1]
```diff
@@ -943,6 +976,7 @@ construct_runtime!(
 		PolkadotXcm: pallet_xcm = 31,
 		CumulusXcm: cumulus_pallet_xcm = 32,
 		// DmpQueue: cumulus_pallet_dmp_queue = 33, removed
+		ToKusamaXcmRouter: pallet_xcm_bridge_hub_router::<Instance1> = 34,
```
[the remark this answers is not in the document]
Comment (hbulgarini) [reply system-parachains/asset-hub-paseo/src/lib.rs:979]: I believe we do. Is much easier for backporting future releases to have the most similar runtime even if the bridge is never activated.

File: system-parachains/asset-hub-paseo/src/lib.rs [hunk 2]
```diff
@@ -827,6 +828,38 @@ impl pallet_nfts::Config for Runtime {
 	type Helper = ();
 }
 
+/// XCM router instance to BridgeHub with bridging capabilities for `Kusama` global
+/// consensus with dynamic fees and back-pressure.
+pub type ToKusamaXcmRouterInstance = pallet_xcm_bridge_hub_router::Instance1;
+impl pallet_xcm_bridge_hub_router::Config<ToKusamaXcmRouterInstance> for Runtime {
```
[the remark this answers is not in the document]
Comment (hbulgarini) [reply system-parachains/asset-hub-paseo/src/lib.rs:834]: I believe we do. Is much easier for backporting future releases to have the most similar runtime even if the bridge is never activated.

File: system-parachains/asset-hub-paseo/src/xcm_config.rs [hunk 3]
```diff
@@ -491,11 +480,98 @@ pub mod bridging {
...
+		parameter_types! {
+			pub SiblingBridgeHubWithBridgeHubKusamaInstance: Location = Location::new(
+				1,
+				[
+					Parachain(SiblingBridgeHubParaId::get()),
+					PalletInstance(bp_bridge_hub_polkadot::WITH_BRIDGE_POLKADOT_TO_KUSAMA_MESSAGES_PALLET_INDEX),
+				]
+			);
+
+			pub const KusamaNetwork: NetworkId = NetworkId::Kusama;
```
[the remark this answers is not in the document]
Comment (hbulgarini) [reply system-parachains/asset-hub-paseo/src/xcm_config.rs:518]: As it's not in the roadmap for the short term to deploy Westend bridge, i think this is an additional overhead where the demand for this is not justified.

**Comment by al3mart** [comment]
`try-runtime` run for asset-hub's runtime looks pretty green as well

```
[2024-09-09T12:27:00Z INFO  remote-ext] since no at is provided, setting it to latest finalized head, 0xdbb4adfdef67f100c0aa852215c6ef56ee5fde9dab00b32b18680e920c409d89
[2024-09-09T12:27:00Z INFO  remote-ext] since no prefix is filtered, the data for all pallets will be downloaded
[2024-09-09T12:27:00Z INFO  remote-ext] scraping key-pairs from remote at block height 0xdbb4adfdef67f100c0aa852215c6ef56ee5fde9dab00b32b18680e920c409d89
✅ Found 962301 keys (27.75s)
[00:00:13] Downloading key values 14,617.6003/s [==============>---------------------------------------------------------] 198340/962301 (52s)[2024-09-09T12:27:41Z WARN  frame_remote_externalities] Batch request failed (2/12 retries). Error: connection error: received fatal alert: BadRecordMac
[00:01:18] ✅ Downloaded key values 12,209.3891/s [=======================================================================] 962301/962301 (0s)
✅ Inserted keys into DB (4.51s)
[2024-09-09T12:28:52Z INFO  remote-ext] adding data for hashed prefix: , took 112.04s
[2024-09-09T12:28:52Z INFO  remote-ext] adding data for hashed key: 3a636f6465
[2024-09-09T12:28:52Z INFO  remote-ext] adding data for hashed key: 26aa394eea5630e07c48ae0c9558cef7f9cce9c888469bb1a0dceaa129672ef8
[2024-09-09T12:28:52Z INFO  remote-ext] adding data for hashed key: 26aa394eea5630e07c48ae0c9558cef702a5c1b19ab7a04f536c519aca4983ac
[2024-09-09T12:28:52Z INFO  remote-ext] 👩‍👦o child roots found to scrape
[2024-09-09T12:28:52Z INFO  remote-ext] initialized state externalities with storage root 0xb3eaed052a03a5bbc0218f85164120e0baac1462b5afebedd2bf5154f10942db and state_version V0
[2024-09-09T12:28:53Z INFO  try-runtime::cli] Original runtime [Name: RuntimeString::Owned("asset-hub-paseo")] [Version: 1002006] [Code hash: 0x894a...182f]
[2024-09-09T12:28:53Z INFO  try-runtime::cli] New runtime      [Name: RuntimeString::Owned("asset-hub-paseo")] [Version: 1002008] [Code hash: 0xc56d...9df8]
[2024-09-09T12:28:53Z INFO  try-runtime::cli] 🚀 Speed up your workflow by using snapshots instead of live state. See `try-runtime create-snapshot --help`.
[2024-09-09T12:28:53Z INFO  try_runtime_core::common::misc_logging] ------------------------------------------------------------------
    
    
[2024-09-09T12:28:53Z INFO  try_runtime_core::common::misc_logging] 🔬 Running TryRuntime_on_runtime_upgrade with checks: PreAndPost
    
    
[2024-09-09T12:28:53Z INFO  try_runtime_core::common::misc_logging] ------------------------------------------------------------------
    
    
[2024-09-09T12:28:57Z INFO  runtime::frame-support] 🐥 New pallet "ToKusamaXcmRouter" detected in the runtime. The pallet has no defined storage version, so the on-chain version is being initialized to StorageVersion(0).
[2024-09-09T12:28:57Z WARN  frame_support::migrations] Migration RemovePallet<DmpQueue> can be removed (no keys found pre-removal).
[2024-09-09T12:28:57Z INFO  frame_support::migrations] Removed 0 DmpQueue keys 🧹
[2024-09-09T12:28:57Z INFO  frame_support::migrations] No DmpQueue keys found post-removal 🎉
[2024-09-09T12:28:57Z WARN  frame_support::migrations] 🚚 Pallet "XcmpQueue" VersionedMigration migration 3->4 can be removed; on-chain is already at StorageVersion(4).
[2024-09-09T12:28:57Z INFO  runtime::collator-selection] Migration did not execute. This probably should be removed
[2024-09-09T12:29:06Z INFO  runtime::executive] ✅ Entire runtime state decodes without error. 47016591 bytes total.
[2024-09-09T12:29:08Z INFO  try_runtime_core::common::misc_logging] ------------------------------------------------------------------------------------------------------
    
    
[2024-09-09T12:29:08Z INFO  try_runtime_core::common::misc_logging] 🔬 TryRuntime_on_runtime_upgrade succeeded! Running it again without checks for weight measurements.
    
    
[2024-09-09T12:29:08Z INFO  try_runtime_core::common::misc_logging] ------------------------------------------------------------------------------------------------------
    
    
[2024-09-09T12:29:08Z INFO  runtime::frame-support] 🐥 New pallet "ToKusamaXcmRouter" detected in the runtime. The pallet has no defined storage version, so the on-chain version is being initialized to StorageVersion(0).
[2024-09-09T12:29:08Z INFO  frame_support::migrations] Removed 0 DmpQueue keys 🧹
[2024-09-09T12:29:08Z WARN  frame_support::migrations] 🚚 Pallet "XcmpQueue" VersionedMigration migration 3->4 can be removed; on-chain is already at StorageVersion(4).
[2024-09-09T12:29:08Z INFO  runtime::collator-selection] Migration did not execute. This probably should be removed
[2024-09-09T12:29:08Z INFO  try_runtime_core::common::misc_logging] ---------------------------------------------------------------------------------
    
    
[2024-09-09T12:29:08Z INFO  try_runtime_core::common::misc_logging] 🔬 Running TryRuntime_on_runtime_upgrade again to check idempotency: PreAndPost
    
    
[2024-09-09T12:29:08Z INFO  try_runtime_core::common::misc_logging] ---------------------------------------------------------------------------------
    
    
[2024-09-09T12:29:08Z WARN  frame_support::migrations] Migration RemovePallet<DmpQueue> can be removed (no keys found pre-removal).
[2024-09-09T12:29:08Z INFO  frame_support::migrations] Removed 0 DmpQueue keys 🧹
[2024-09-09T12:29:08Z INFO  frame_support::migrations] No DmpQueue keys found post-removal 🎉
[2024-09-09T12:29:08Z WARN  frame_support::migrations] 🚚 Pallet "XcmpQueue" VersionedMigration migration 3->4 can be removed; on-chain is already at StorageVersion(4).
[2024-09-09T12:29:08Z INFO  runtime::collator-selection] Migration did not execute. This probably should be removed
[2024-09-09T12:29:17Z INFO  runtime::executive] ✅ Entire runtime state decodes without error. 47016591 bytes total.
[2024-09-09T12:29:19Z INFO  try-runtime::cli] PoV size (zstd-compressed compact proof): 3.2 KB. For parachains, it's your responsibility to verify that a PoV of this size fits within any relaychain constraints.
[2024-09-09T12:29:19Z INFO  try-runtime::cli] Consumed ref_time: 0.00095s (0.19% of max 0.5s)
[2024-09-09T12:29:19Z INFO  try-runtime::cli] ✅ No weight safety issues detected. Please note this does not guarantee a successful runtime upgrade. Always test your runtime upgrade with recent state, and ensure that the weight usage of your migrations will not drastically differ between testing and actual on-chain execution.
```

File: system-parachains/bridge-hub-paseo/src/lib.rs [hunk 4]
```diff
@@ -159,20 +172,20 @@ impl_opaque_keys! {
...
 pub const VERSION: RuntimeVersion = RuntimeVersion {
-    spec_name: create_runtime_str!("bridge-hub-polkadot"),
-    impl_name: create_runtime_str!("bridge-hub-polkadot"),
-    authoring_version: 1,
-    spec_version: 1_001_006,
-    impl_version: 0,
-    apis: RUNTIME_API_VERSIONS,
-    transaction_version: 4,
-    state_version: 1,
+	spec_name: create_runtime_str!("bridge-hub-paseo"),
```
Comment (al3mart) [inline system-parachains/bridge-hub-paseo/src/lib.rs:175]: Unfortunately bridge hub was launch with the spec name `bridge-hub-polkadot`.
To keep this change in we have to be careful of upgrading the network by approving the new code with `system.authorizeUpgradeWithoutChecks`.

Nothing to block the changes from merging, but something to have into consideration when it comes to enact this new code.

**Review by al3mart — APPROVED** [review]

**Pull request merged**